lib/model.js

```diff
@@ -4424,11 +4424,15 @@ function _execPopulateQuery(mod, match, select, assignmentOpts, callback) {
     limit: mod.options.limit,
     perDocumentLimit: mod.options.perDocumentLimit
   }, mod.options.options);
+
   if (mod.count) {
     delete queryOptions.skip;
   }
 
-  if (queryOptions.perDocumentLimit != null) {
+  if (queryOptions.perDocumentLimit != null && queryOptions.limit != null) {
```

Comment:
I think the better place to put this might be `lib/options/PopulateOptions.js`. That should make it easier - checking options in `populate()` can get confusing.